src/frontend/src/pages/HomePage/components/TimelineSection.tsx

```diff
@@ -1,8 +1,59 @@
-import { Typography } from '@mui/material';
+import { Grid, Typography } from '@mui/material';
+import * as React from 'react';
+import Timeline from '@mui/lab/Timeline';
+import TimelineItem from '@mui/lab/TimelineItem';
+import TimelineSeparator from '@mui/lab/TimelineSeparator';
+import TimelineConnector from '@mui/lab/TimelineConnector';
+import TimelineContent from '@mui/lab/TimelineContent';
+import TimelineDot from '@mui/lab/TimelineDot';
+import { useAllMilestones } from '../../../hooks/recruitment.hooks';
+import LoadingIndicator from '../../../components/LoadingIndicator';
+import ErrorPage from '../../ErrorPage';
+import dayjs from 'dayjs';
 
 const TimelineSection = () => {
-  // TODO: Ticket #2715
-  return <Typography>Timeline Section</Typography>;
+  const { isLoading, isError, error, data: mileStones } = useAllMilestones();
+
+  if (isLoading || !mileStones) return <LoadingIndicator />;
+  if (isError) return <ErrorPage error={error} message={error.message} />;
+
+  const isPastEvent = (date: Date) => dayjs(date).isBefore(dayjs());
+
+  const getDotColor = (date: Date) => (isPastEvent(date) ? 'primary' : 'grey');
+  const getConnectorStyle = (date: Date) => ({
+    height: {
+      xs: '100px',
+      sm: '150px',
+      md: '200px'
+    },
+    backgroundColor: isPastEvent(date) ? 'primary.main' : 'grey'
+  });
+
+  const fillLine = () => {};
+
+  return (
+    <Grid>
+      <Timeline position="alternate">
+        {mileStones.map((milestone, index) => (
+          <TimelineItem key={index}>
+            <TimelineSeparator>
+              <TimelineDot color={getDotColor(milestone.dateOfEvent)} sx={{ width: '20px', height: '20px' }} />
```

Comment:
instead of using px, try % or vh/vw, this way when you shrink the page, it shoudl adjust as well